qualcommax: add missing WAN LED support to Spectrum SAX1V1K routers

iv7777 opened the pull request:
Fixed an issue where both WAN LEDs light up before plugging in the ethernet cable and no blinking regardless of WAN network activity. Updated the LED configuration to reflect proper status:
- Green indicates connection speed.
- Yellow indicates traffic activity.

This resolves inconsistent WAN LED behavior on Spectrum SAX1V1K routers.

Signed-off-by:  Ivan Deng <hongba@rocketmail.com>

musashino205 commented on target/linux/qualcommax/ipq807x/base-files/etc/board.d/01_leds:
```diff
@@ -16,6 +16,9 @@ asus,rt-ax89x)
 	ucidef_set_led_netdev "sfp" "SFP" "white:sfp" "10g-sfp"
 	ucidef_set_led_netdev "wan" "WAN" "white:wan" "wan"
 	;;
+spectrum,sax1v1k)
+	ucidef_set_led_netdev "wan-port-link-yellow" "WAN-PORT-LINK-YELLOW" "90000.mdio-1:1c:yellow:wan" "wan" "tx rx link_10 link_100 link_1000 link_2500"
+	;;
```
Please sort alphabetically.

iv7777 replied:
It has been changed.

musashino205 commented:
Please,

- use `qualcommax: ` as a prefix of the commit title
- add missing detailed description in the commit
- add missing `Signed-off-by: ` line in the commit description
- use your real name for the `Signed-off-by: ` line and the [author filed (`From: `)](https://patch-diff.githubusercontent.com/raw/openwrt/openwrt/pull/17623.patch)

After updating, please don't close this PR or delete the branch, and use `git push -f` to update this PR instead.

see details:

- https://openwrt.org/submitting-patches#submission_guidelines
- https://openwrt.org/docs/guide-developer/working-with-github-pr
- https://openwrt.org/docs/guide-developer/frequent-pr-mistakes

iv7777 committed:
qualcommax: add missing WAN LED support to Spectrum SAX1V1K routers

Fixed an issue where both WAN LEDs light up before plugging in the
ethernet cable and no blinking regardless of WAN network activity.

Updated the LED configuration to reflect proper status:
Green indicates 2.5Gb connection speed.
Yellow indicates other connection speed and traffic activity.

This resolves inconsistent WAN LED behavior on Spectrum SAX1V1K routers.

Signed-off-by: Ivan Deng <hongba@rocketmail.com>
Link: https://github.com/openwrt/openwrt/pull/17623
Signed-off-by: Robert Marko <robimarko@gmail.com>

iv7777 commented:
Commit updated based on the advised rules.

robimarko commented:
Please fix the formal issues, email in SoB needs to be inside of `<>`

robimarko commented on target/linux/qualcommax/ipq807x/base-files/etc/board.d/01_leds:
```diff
@@ -41,6 +41,9 @@ netgear,wax630)
 	ucidef_set_led_netdev "wan" "WAN" "blue:network" "wan"
 	;;
+spectrum,sax1v1k)
+	ucidef_set_led_netdev "wan-port-link-yellow" "WAN-PORT-LINK-YELLOW" "90000.mdio-1:1c:yellow:wan" "wan" "tx rx link_10 link_100 link_1000 link_2500"
```
Green LED is not configured?

iv7777 replied:
here's the ls command output
```
root@openwrt:/# ls /sys/class/leds/90000.mdio-1:1c:green:wan
brightness      max_brightness  subsystem       uevent
device          power           trigger

root@openwrt:/# ls /sys/class/leds/90000.mdio-1:1c:yellow:wan
brightness      half_duplex     link_100        link_5000       rx              uevent
device          interval        link_1000       max_brightness  subsystem
device_name     link            link_10000      offloaded       trigger
full_duplex     link_10         link_2500       power           tx
```

robimarko replied:
Are you sure that both LED-s actually exist and are connected to the PHY?

You can easily test whether its connected to the PHY by toggling its brightness manually via sysfs

iv7777 replied:
For the yellow one, yes, I have full control over it. However, the green one is weird. Setting the configuration of 'active-low' in the DTS file can actually turn it off by default, but there is no control over anything else whatsoever. In the stock firmware, the green one is fully functional, including blinking and speed indication. Not sure it has different wiring or not than Dynalink DL-WRX36 for the WAN LEDs. Hope someone with more in-depth knowledge can sort this out.

robimarko replied:
@iv7777 Dynalink LED config was actually incorrect, I fixed it today:
https://github.com/openwrt/openwrt/commit/2d077913cd53cad33123c8a9bf6da8a1c5576c69

So try that

iv7777 replied:
yes, tested works. have to keep 'active-low' to the green led configuration to turn it off when no cable is plugged in. wondering if this is the case for DL-WRX36 too. I am updating the PR now.

musashino205 commented on target/linux/qualcommax/ipq807x/base-files/etc/board.d/01_leds:
```diff
@@ -41,6 +41,10 @@ netgear,wax630)
 redmi,ax6)
 	ucidef_set_led_netdev "wan" "WAN" "blue:network" "wan"
 	;;
+spectrum,sax1v1k)
+	ucidef_set_led_netdev "wan-port-link-green" "WAN-PORT-LINK-GREEN" "90000.mdio-1:1c:green:wan" "wan" "link_2500"
+	ucidef_set_led_netdev "wan-port-link-yellow" "WAN-PORT-LINK-YELLOW" "90000.mdio-1:1c:yellow:wan" "wan" "tx rx link_10 link_100 link_1000"
+	;;
```
This configuration is the same as `dynalink,dl-wrx36` now, so those can be merged.

iv7777 replied:
Pls help to merge

iv7777 replied:
Pls help to merge.

robimarko replied:
Just add the `spectrum,sax1v1k` to the same case as `dynalink,dl-wrx36`

robimarko replied:
I did the changes, quite literaly all you had to do is add spectrum device to the dynalink case

robimarko commented:
Thanks! Rebased on top of main and merged!
